packages/experimental/Switch/src/Switch.types.ts

```diff
@@ -168,17 +173,21 @@ export interface SwitchProps extends Omit<IWithPressableOptions<ViewProps>, 'onP
 
 export type SwitchState = IPressableState & { toggled?: boolean };
 
+export type textBeingTestedStates = 'init' | 'onText' | 'offText' | 'done';
+
 export interface SwitchInfo {
   props: IWithPressableEvents<SwitchProps & React.ComponentPropsWithRef<any>>;
   state: SwitchState;
 }
 
+export type toggleContainerProps = IViewProps & { ref: React.RefObject<View> };
```

Comment:
>IViewProps [](http://example.com/codeflow?start=35&length=10)

Does `React.PropsWithRef<IViewProps>;` not work here?

Reply:
No it doesn't. When I try it I get the error `Type 'IViewProps' does not satisfy the constraint 'ElementType<any>`

Reply:
I get the error `Type 'MutableRefObject<any>' has no properties in common with type 'IViewProps'` when I add it.

Reply:
Changed implementation.